src/game/AI/ScriptDevAI/scripts/northrend/ulduar/ulduar/boss_ignis.cpp

```diff
@@ -73,309 +78,295 @@ enum
 
     // NPC ids
     NPC_SCORCH                          = 33221,
-    NPC_IRON_CONSTRUCT                  = 33121,        // constructs which are activated on demand by Ignis
 
     MAX_HEAT_STACKS                     = 10,
+
+    // stringIds
+    ULDUAR_IGNIS_ADDS                   = 6030101,
 };
 
 /*######
 ## boss_ignis
 ######*/
 
-struct boss_ignisAI : public ScriptedAI
+enum IgnisActions
+{
+    IGNIS_ACTION_MAX,
+};
+
+struct boss_ignisAI : public BossAI
 {
-    boss_ignisAI(Creature* pCreature) : ScriptedAI(pCreature)
+    boss_ignisAI(Creature* creature) : BossAI(creature, IGNIS_ACTION_MAX),
+    m_instance(static_cast<instance_ulduar*>(creature->GetInstanceData())),
+    m_isRegularMode(creature->GetMap()->IsRegularDifficulty())
     {
-        m_pInstance = (instance_ulduar*)pCreature->GetInstanceData();
-        m_bIsRegularMode = pCreature->GetMap()->IsRegularDifficulty();
-        Reset();
+        SetDataType(TYPE_IGNIS);
+        AddOnAggroText(SAY_AGGRO);
+        AddOnKillText(SAY_SLAY_1, SAY_SLAY_2);
+        AddOnDeathText(SAY_DEATH);
+        AddCastOnDeath({ObjectGuid(), SPELL_KILL_ALL_CONSTRUCTS, TRIGGERED_OLD_TRIGGERED});
+        AddRespawnOnEvade(30s);
     }
 
-    instance_ulduar* m_pInstance;
-    bool m_bIsRegularMode;
+    instance_ulduar* m_instance;
+    bool m_isRegularMode;
 
-    uint32 m_uiBerserkTimer;
-    uint32 m_uiFlameJetsTimer;
-    uint32 m_uiSlagPotTimer;
-    uint32 m_uiScorchTimer;
-    uint32 m_uiConstructTimer;
-
-    void Reset() override
+    void JustRespawned() override
     {
-        m_uiBerserkTimer        = 10 * MINUTE * IN_MILLISECONDS;
-        m_uiFlameJetsTimer      = 20000;
-        m_uiSlagPotTimer        = 25000;
-        m_uiScorchTimer         = 13000;
-        m_uiConstructTimer      = 10000;
+        BossAI::JustRespawned();
+        m_creature->GetMap()->GetVariableManager().SetVariable(WORLD_STATE_CUSTOM_RESPAWN_IGNIS_ADDS, 0);
     }
 
-    void JustDied(Unit* /*pKiller*/) override
+    void EnterEvadeMode() override
     {
-        if (m_pInstance)
-            m_pInstance->SetData(TYPE_IGNIS, DONE);
-
-        DoCastSpellIfCan(m_creature, SPELL_KILL_ALL_CONSTRUCTS, CAST_TRIGGERED);
-        DoScriptText(SAY_DEATH, m_creature);
+        BossAI::EnterEvadeMode();
+        auto* ignisAdds = m_creature->GetMap()->GetCreatures(ULDUAR_IGNIS_ADDS);
```

Comment:
If its a pointer, add a guard. Someone will nuke some db data and core will crash.